test/clj/rems/db/test_applications.clj

```diff
@@ -319,3 +324,13 @@
                                            :catalogue-item-ids [cat-id2]
                                            :time (DateTime. 1000)
                                            :actor "alice"})))))))
+
+(deftest test-allocate-external-id!
+  (is (= [] (db/get-external-ids {:prefix "a"})))
+  (is (= [] (db/get-external-ids {:prefix "b"})))
+  (is (= {:prefix "a" :suffix "1"} (allocate-external-id! "a")))
+  (is (= {:prefix "a" :suffix "2"} (allocate-external-id! "a")))
+  (is (= {:prefix "a" :suffix "3"} (allocate-external-id! "a")))
+  (is (= {:prefix "b" :suffix "1"} (allocate-external-id! "b")))
+  (is (= {:prefix "b" :suffix "2"} (allocate-external-id! "b")))
+  (is (= {:prefix "a" :suffix "4"} (allocate-external-id! "a"))))
```

Comment:
If we are going to use year as the prefix at the moment, would maybe be consistent to give prefix in the same example format?